app/models/concerns/foreman_wreckingball/host_extensions.rb

```diff
@@ -31,6 +31,8 @@ module HostExtensions
               :foreign_key => 'host_id',
               :inverse_of => :host,
               :dependent => :destroy
+
+      scope :owned_by_current_user, -> { where("owner_type = 'User' AND owner_id = ?", User.current.id).or(where("owner_type = 'Usergroup' AND owner_id = ?", User.current.usergroups.ids)) }
```

Comment:
I'm wondering if we need to make this work recursively. A usergroup can contain child usergroups.

Reply:
Can this be written like this: `where(owner_type: 'User', owner_id: User.current.id)` ?

Reply:
> I'm wondering if we need to make this work recursively. A usergroup can contain child usergroups.

good point

Reply:
And... this currently gives me the following error:

```
11:18:13 rails.1   | 2018-11-20T11:18:13 [D|sql|ea5b4]    (0.5ms)  SELECT COUNT(host_status.status) AS count_host_status_status, host_status.status AS host_status_status FROM "hosts" INNER JOIN "host_status" ON "host_status"."host_id" = "hosts"."id" AND "host_status"."type" IN ('ForemanWreckingball::ToolsStatus') WHERE "hosts"."type" IN ('Host::Managed') AND ((owner_type = 'User' AND owner_id = 3) OR (owner_type = 'Usergroup' AND owner_id = 4,2,3)) GROUP BY host_status.status
11:18:13 rails.1   | 2018-11-20T11:18:13 [D|sql|ea5b4]   ↳ /Users/f0218175/foremandev/foreman_wreckingball/app/controllers/foreman_wreckingball/hosts_controller.rb:39
11:18:13 rails.1   | 2018-11-20T11:18:13 [I|app|ea5b4] Completed 500 Internal Server Error in 23ms (ActiveRecord: 2.9ms)
11:18:13 rails.1   |  |
11:18:13 rails.1   |  |
11:18:13 rails.1   | 2018-11-20T11:18:13 [F|app|]
11:18:13 rails.1   | 2018-11-20T11:18:13 [F|app|] ActiveRecord::StatementInvalid (PG::DatatypeMismatch: ERROR:  argument of OR must be type boolean, not type record
11:18:13 rails.1   |  | LINE 1: ...') AND ((owner_type = 'User' AND owner_id = 3) OR (owner_typ...
11:18:13 rails.1   |  |                                                              ^
11:18:13 rails.1   |  | : SELECT COUNT(host_status.status) AS count_host_status_status, host_status.status AS host_status_status FROM "hosts" INNER JOIN "host_status" ON "host_status"."host_id" = "hosts"."id" AND "host_status"."type" IN ('ForemanWreckingball::ToolsStatus') WHERE "hosts"."type" IN ('Host::Managed') AND ((owner_type = 'User' AND owner_id = 3) OR (owner_type = 'Usergroup' AND owner_id = 4,2,3)) GROUP BY host_status.status):
11:18:13 rails.1   | 2018-11-20T11:18:13 [F|app|]
11:18:13 rails.1   | 2018-11-20T11:18:13 [F|app|] /Users/f0218175/foremandev/foreman_wreckingball/app/controllers/foreman_wreckingball/hosts_controller.rb:39:in `block in status_dashboard'
11:18:13 rails.1   |  | /Users/f0218175/foremandev/foreman_wreckingball/app/controllers/foreman_wreckingball/hosts_controller.rb:29:in `map'
11:18:13 rails.1   |  | /Users/f0218175/foremandev/foreman_wreckingball/app/controllers/foreman_wreckingball/hosts_controller.rb:29:in `status_dashboard'
11:18:13 rails.1   |  | app/controllers/concerns/application_shared.rb:15:in `set_timezone'
11:18:13 rails.1   |  | app/models/concerns/foreman/thread_session.rb:32:in `clear_thread'
11:18:13 rails.1   |  | app/controllers/concerns/foreman/controller/topbar_sweeper.rb:12:in `set_topbar_sweeper_controller'
11:18:13 rails.1   |  | lib/middleware/telemetry.rb:9:in `call'
11:18:13 rails.1   |  | lib/middleware/catch_json_parse_errors.rb:8:in `call'
11:18:13 rails.1   |  | lib/middleware/logging_context.rb:23:in `call'
```